test/middleware/spec.back-link.js

```diff
@@ -1,11 +1,20 @@
-var backLinks = require('../../lib/middleware/back-links');
-
-var util = require('util'),
+var proxyquire = require('proxyquire'),
+    util = require('util'),
     EventEmitter = require('events').EventEmitter;
 
+var getBackLinks = function (previousSteps) {
+    return proxyquire('../../lib/middleware/back-links', {
+        '../util/helpers': {
+            getPreviousSteps: function () {
+                return previousSteps;
+            }
+        }
+    });
+};
```

Comment:
we could do
```js
var getBackLinks = function (previousSteps) {
    return proxyquire('../../lib/middleware/back-links', {
        '../util/helpers': {
            getPreviousSteps: sinon.sub().returns(previousSteps)
        }
    });
};
```
but we need to specify the steps it will return to properly test the functionality of the middleware